source/views/guides/configuring-diesel-cli.html.slim

```diff
@@ -0,0 +1,119 @@
+          worry about running a separate command to update your Rust code. It is
+          highly recommended that you use this field, to ensure that the Rust
+          representation of your database schema is always in sync with what is
+          actually in your database. Typically this is set to `src/schema.rs`.
+
+          Unlike other fields, this doesn't actually modify the behavior of
+          `diesel print-schema`. `diesel print-schema` will always output your
+          schema to stdout, regardless of whether this field is present or not.
+
+          The `with_docs` field
+          ---------------------
+
+          When this field is set to `true`, `diesel print-schema` will act as
+          though the `--with-docs` flag was passed by default. This places a doc
+          comment on all tables and columns.
+
+          The `filter` field
+          ------------------
+
+          This field specifies which tables should be output by `diesel
+          print-schema`. It corresponds to the `--only-tables` and
+          `--except-tables` on the command line. Its value should be a map with
+          one of those two keys. For example:
+
+        .demo__example
+          .demo__example-browser
+            .browser-bar diesel.toml
+            pre.demo__example-snippet
+              code
+                | [print-schema]
+                  # This will cause only the users and posts tables to be output
+                  filter = { only_tables = ["users", "posts"] }
+
+                  # This will cause all tables *except* the comments table to be
+                  # output
+                  filter = { except_tables = ["comments"] }
+
+        markdown:
+          The `schema` field
+          ------------------
+
+          Specifies which schema to use when searching for tables. When set,
+          `diesel print-schema` will always behave as though `--schema` were
+          passed. This field only affects PostgreSQL. If no value is provided,
+          the `public` schema will be searched.
+
+          The `import_types` field
+          ------------------------
+
+          This field adds `use` statements to the top of every `table!`
+          declaration. When set, `diesel print-schema` will behave as if
+          `--import-types` were passed. When no value is given, only types from
+          `diesel::sql_types` will be imported.
+
+        .demo__example
+          .demo__example-browser
+            .browser-bar diesel.toml
+            pre.demo__example-snippet
+              code
+                | [print-schema]
+                  # Add types from `diesel_full_text_search` like `tsvector`
+                  import_types = ["diesel::sql_types::*", "diesel_full_text_search::types::*"]
+
+        markdown:
+          The `patch_file` field
+          ------------------------
+
+          Specifies a `.patch` file to be applied to your schema after it is
```

Comment:
Is the schema generation 100% deterministic, especially around ordering?

Reply:
Yes. Tables are ordered alphabetically. Columns are ordered by definition order.